### ISSUE TITLE
Receiver type checks for `super` invocations

### ISSUE THREAD

**Issue by wmdietl** [typetools/checker-framework#2263]
Take the following simple example:

```java
import org.checkerframework.checker.initialization.qual.UnderInitialization;

class Super {
    void foo(@UnderInitialization(Super.class) Super this) {}
}

class Sub extends Super {
    @Override
    void foo(@UnderInitialization(Object.class) Sub this) {
        super.foo();
    }
}
```

This incorrectly passes the Nullness Checker.
The invocation `super.foo()` has an invalid receiver type and an error should be raised.

**Comment by aagarwal1012** [comment]
@wmdietl, I want to work on this issue. Can you please guide me?

**Comment by t-rasmud** [comment]
@wmdietl When I tried to reproduce this issue, I got the expected error. See PR https://github.com/typetools/checker-framework/pull/2308
